### PR TITLE
[Reviewer: Ellie] Astaire alarm log updates

### PR DESCRIPTION
**Pull request by johadalin**
I'd particularly like you to look at the UTs. I've completely replaced what was there before. The diff below makes it look like i edited them about, but they are just all new, trying to test the error states rather than using the 'alarmed' check.

### CODE REVIEW

**Comment by johadalin** [comment]
Cpp- common changes are here: https://github.com/Metaswitch/cpp-common/pull/444

**Comment by johadalin** [comment]
Astaire changes here: https://github.com/Metaswitch/astaire/pull/44

File: src/main.cpp [hunk 1]
```diff
@@ -2076,9 +2062,8 @@ int main(int argc, char* argv[])
 
     if (local_data_store == NULL)
     {
-      CL_SPROUT_MEMCACHE_CONN_FAIL.log();
```
Comment (eleanor-merry) [inline src/main.cpp]: What is this condition? Why doesn't it get an ENT log?

File: src/ut/communicationmonitor_test.cpp [hunk 2]
```diff
@@ -68,54 +68,129 @@ class CommunicationMonitorTest : public ::testing::Test
   CommunicationMonitor _cm;
 };
 
-TEST_F(CommunicationMonitorTest, SuccessNoUpdate)
+// State Definitions:
+//
+// NO_ERRORS == No failures, at least one success
```
Comment (eleanor-merry) [inline src/ut/communicationmonitor_test.cpp]: Can you explain more about what the test is doing here, rather than using the state definitions

**Comment by eleanor-merry** [comment]
Looks good.

**Pull request merged**